org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java

```diff
@@ -441,14 +441,20 @@ public static boolean matches(Pattern p, CharSequence input) {
 	public Map<String, String> getTextSubstitutions(String inputFileName) {
 		String simpleFileName = FileUtils.getFileNameFromFullyQualifiedFileName(inputFileName);
 
-		Map<String, String> specificUpdates = getSpecificTextUpdates().get(simpleFileName);
+		Map<String, Map<String, String>> specificUpdates = getSpecificTextUpdates();
 		if (specificUpdates != null) {
-			return specificUpdates;
+			Map<String, String> updates = specificUpdates.get(simpleFileName);
+			if (updates != null) {
+				return updates;
+			}
 		}
 
-		for (Map.Entry<Pattern, Map<String, String>> wildcardEntry : getWildCardTextUpdates().entrySet()) {
-			if (matches(wildcardEntry.getKey(), simpleFileName)) {
-				return wildcardEntry.getValue();
+		Map<Pattern, Map<String, String>> wildcardUpdates = getWildCardTextUpdates();
```

Comment:
Wouldn't it be better if `getWildCardTextUpdates` never returned null? At least an empty map. Having these null checks is not the best.

Reply:
Updated to answer an empty map.